### ISSUE TITLE
2 separate pools cause issue.

### ISSUE THREAD

**Issue by ericlovescode** [hgourvest/node-firebird#39]
I have 2 main parts to my application so I want to create 2 separate pools. If I give the 1st pool 5 connections then I can only give the 2nd pool maximum 3 connections or it will die with the following message:

Lender V3DATA connections: 5
Product V3DATA connections: 4
Express server listening on port 3000
V3DATA connected (Settings)...
V3DATA connected (Download)...
(node) warning: possible EventEmitter memory leak detected. 11 listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at Database.addListener (events.js:160:15)
    at Connection.attach (/home/eric/local/node/jstrap/node_modules/node-firebird/lib/index.js:1781:21)
    at /home/eric/local/node/jstrap/node_modules/node-firebird/lib/index.js:1248:21
    at doCallback (/home/eric/local/node/jstrap/node_modules/node-firebird/lib/index.js:915:5)
    at Socket.<anonymous> (/home/eric/local/node/jstrap/node_modules/node-firebird/lib/index.js:1536:17)
    at Socket.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:764:14)
    at Socket.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:426:10)
    at emitReadable (_stream_readable.js:422:5)

I'm using the latest node-firebird release. Cheers.

**Comment by eddyvinsensius** [comment]
> Multiple queries don't work in one DB context, because the socket can handle only one statement at time. The pooling is very good for web apps and multiple requests, but each request is handled quasi synchronous (query1 --> end --> query2 --> end --> query3 --> end --> detach).
> 
> BTW: you can execute multiple queries like this: https://github.com/hgourvest/node-firebird/blob/master/test/run.js#L73
> 
> Thanks.

hi sir could you kindly help me repost this link because it give me error 404 whenever i click it , thank's alot sir

**Comment by eddyvinsensius** [comment]
i have a code like this, it run but sometimes got freeze so i have to restart the program, i have tried classic & pooling but both give me the same problem , pls help me sir :

var tSql=JSON.parse(sql);
// Firebird.attach(options, (err, db) => {
var pool = Firebird.pool(20, options);
pool.get( (err, db) => {    
    if (err) {return res.status(400).send(`\n${err.message}\n`);}

    if (tSql.length==1){
        db.query(tSql[0]["PID"], (err, data) => {
            if (err) {
                db.detach();
                return res.status(400).send(`\n${err.message}\n`);
            }
            db.detach();
            return res.send([{"STATUS": "p#002a .."}]);
        });
    } else {
        db.transaction(Firebird.ISOLATION_READ_COMMITED, function(err, transaction) {
            for (let i = 0; i < tSql.length; i++) {
                transaction.query(tSql[i]["PID"], (err, data) => {
                    if (err) {
                        transaction.rollback();
                        db.detach();
                        return res.status(400).send(`\n${err.message}\n`);
                    }
                });
            }    

            transaction.query('select first 1 id from fsys', (err, data) => {
                if (err) {
                    transaction.rollback();
                    db.detach();
                    return res.status(400).send(`\n${err.message}\n`);
                }
                transaction.commit(function(err) {
                    if (err) {
                        transaction.rollback();
                        db.detach();
                        return res.status(400).send(`\n${err.message}\n`);
                    }
                    else {
                        db.detach();
                        return res.send([{"STATUS": "p#002 .."}]);
                    }    
                });
            });                    
        });
    }
});
pool.destroy();
